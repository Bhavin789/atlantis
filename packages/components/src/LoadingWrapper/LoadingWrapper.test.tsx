import React from "react";
import renderer from "react-test-renderer";
import { cleanup, fireEvent, render } from "@testing-library/react";
import { LoadingWrapper } from ".";

afterEach(cleanup);

it.skip("renders a LoadingWrapper", () => {
  const tree = renderer.create(<LoadingWrapper text="Foo" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it.skip("renders a loud LoadingWrapper", () => {
  const tree = renderer
    .create(<LoadingWrapper text="Foo" loud={true} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test.skip("it should call the handler with the new value", () => {
  const clickHandler = jest.fn();
  const text = "Foo";
  const { getByText } = render(
    <LoadingWrapper onClick={clickHandler} text={text} />,
  );

  fireEvent.click(getByText(text));
  expect(clickHandler).toHaveBeenCalled();

  // E.g. If you need a change event, rather than a click event:
  //
  // fireEvent.change(getByLabelText(placeholder), {
  //   target: { value: newValue },
  // });
  // expect(changeHandler).toHaveBeenCalledWith(newValue);
});
