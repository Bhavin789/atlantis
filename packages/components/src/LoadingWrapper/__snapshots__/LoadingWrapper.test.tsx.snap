// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a LoadingWrapper 1`] = `
<div
  className="loadingWrapper"
/>
`;

exports[`renders a loud LoadingWrapper 1`] = `
<div
  className="loadingWrapper"
/>
`;
